src/librustc/hir/map/collector.rs

```diff
@@ -27,6 +27,9 @@ pub struct NodeCollector<'ast> {
     pub map: Vec<MapEntry<'ast>>,
     /// The parent of this node
     pub parent_node: NodeId,
+    /// Whether to completely ignore nested items,
+    /// such as in decoded HIR fragments from metadata
+    pub ignore_nested_items: bool
```

Comment:
Yeah, it's easier to leave in a bunch of `ItemId`'s that don't point to anything than actually fold them away. As for it being creepy, yes, I just don't want to spend a lot of time on infrastructure that I want removed ASAP. The comment change sounds good.